Sources/XMLCoder/Auxiliaries/XMLElement.swift

```diff
@@ -0,0 +1,280 @@
+            case .unsignedInteger(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .floatingPoint(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .string(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .array(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .dictionary(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            }
+        }
+        
+        if let parentElement = parentElement, let key = key {
+            parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+        }
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: NullBox) {
+        let element = _XMLElement(key: key)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: BoolBox) {
+        let element = _XMLElement(key: key, value: object.xmlString)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: DecimalBox) {
+        let element = _XMLElement(key: key, value: object.xmlString)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: SignedIntegerBox) {
+        let element = _XMLElement(key: key, value: object.xmlString)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: UnsignedIntegerBox) {
+        let element = _XMLElement(key: key, value: object.xmlString)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: FloatingPointBox) {
+        let element = _XMLElement(key: key, value: object.xmlString)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: StringBox) {
+        let element = _XMLElement(key: key, value: object.xmlString)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: ArrayBox) {
+        let objects = object.unbox()
+        
+        for box in objects {
+            switch box {
```

Comment:
Same comment as above, a `protocol` would work better than a `switch` in my opinion.